### PR TITLE
refactor: git and config logic to separate file

### PR DESCRIPTION
**Pull request by nicksnyder**
factored out of https://github.com/sourcegraph/sourcegraph-vscode/pull/15

### CODE REVIEW

**Review by emidoots — APPROVED** [review]
Nice 🚢

File: src/config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict'
```
Comment (nicksnyder) [inline src/config.ts:1]: @slimsag @felixfbecker is this useful/necessary?

Comment (emidoots) [reply src/config.ts:1]: https://stackoverflow.com/questions/31391760/use-strict-needed-in-a-typescript-file

We should enable the always strict TS compiler flag I guess

**Comment by emidoots** [comment]
Merging

File: src/config.ts (hunk 1 above)
Comment (nicksnyder) [reply src/config.ts:1]: https://github.com/sourcegraph/sourcegraph-vscode/pull/17

**Pull request merged**